jerry-core/parser/js/js-parser.c

```diff
@@ -2804,6 +2804,11 @@ parser_parse_script (const uint8_t *arg_list_p, /**< function argument list */
   }
 #endif /* JERRY_DEBUGGER */
 
+  if (JERRY_UNLIKELY (source_size == 0))
```

Comment:
I strongly disagree. In case of source_size == 0, any source_p is valid, since you don't actually read the buffer. In the latter case you read the buffer, and checking buffer readabilaty is beyond the scope of the project.